numcodecs/compat_ext.pxd

```diff
@@ -1,12 +1,11 @@
 # cython: language_level=3
 
 
-cdef class Buffer:
-    cdef:
-        char *ptr
-        Py_buffer buffer
-        size_t nbytes
-        size_t itemsize
-        bint acquired
-
-    cpdef release(self)
+cdef extern from *:
+    """
+    #define PyBytes_RESIZE(b, n) _PyBytes_Resize(&b, n)
+    """
+    int PyBytes_RESIZE(object b, Py_ssize_t n) except -1
```

Comment:
Have added this function (macro) to help with resizing `bytes` objects in Cython

This allows us to truncate `bytes` allocations that are larger than we end up needing without needing to copy to a new `bytes` object